### PR TITLE
Metadata tutorial

### PR DESCRIPTION
**Pull request by hcaseyal**
Simple tutorial for add custom metadata

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,026,842      Total (=)      2,026,842

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,324,728      Total (<)     11,324,732

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,026,842      Total (=)      2,026,842

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,324,726      Total (>)     11,324,722

 No significant differences in binary sizes


```

File: examples/cpp/metadata/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+"Client received message: Hello World"
+
+
+And in the server terminal:
+
+"Header key: custom-bin , value: "
+"Header key: custom-header , value: Custom Value"
+"Header key: user-agent , value: grpc-c++/1.16.0-dev grpc-c/6.0.0-dev (linux; chttp2; gao)"
+
+Note that the value for custom-bin doesn't print nicely because it's a binary
```
Comment (vjpai) [inline examples/cpp/metadata/README.md]: Can you print the binary numbers in hex format, byte by byte? I think the code I pasted below should work....

File: examples/cpp/metadata/greeter_server.cc [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+class GreeterServiceImpl final : public Greeter::Service {
+  Status SayHello(ServerContext* context, const HelloRequest* request,
+                  HelloReply* reply) override {
+    std::string prefix("Hello ");
+
+    // Get the client's initial metadata
+    std::cout << "Client metadata: " << std::endl;
+    const std::multimap<grpc::string_ref, grpc::string_ref> metadata = context->client_metadata();
+    for (auto iter = metadata.begin(); iter != metadata.end(); ++iter) {
+      std::cout << "Header key: " << iter->first << " , value: " << iter->second << std::endl;
```
Comment (vjpai) [inline examples/cpp/metadata/greeter_server.cc]:
``` cpp
std::cout << "Header key: " << iter->first << " , value: ";
size_t isbin = iter->first.find("-bin");
if ((isbin != std::string::npos) && (isbin + 4 == iter->first.size())) {
  std::cout <<  std::hex;
  for (auto c : iter->second) {
    std::cout << static_cast<unsigned int>(c);
  }
  std::cout <<  std::dec;
} else {
  std::cout << iter->second;
}
std::cout << std::endl;
```

**Comment by vjpai** [comment]
One more request: can you make a bazel BUILD file for the example? I know that we actually test examples with bazel but not sure about with Makefile

File: examples/cpp/metadata/greeter_server.cc (hunk 2 above)
Comment (vjpai) [reply examples/cpp/metadata/greeter_server.cc]: I was just trying to convince myself....

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,026,842      Total (=)      2,026,842

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,324,729      Total (<)     11,324,730

 No significant differences in binary sizes


```

**Pull request merged**